core/modules/rproc/InfileMerger.h

```diff
@@ -157,9 +161,9 @@ class InfileMerger {
     std::mutex _mysqlMutex;
     lsst::qserv::mysql::LocalInfile::Mgr _infileMgr;
 
-    // The limited size pool will keep this query from using up all the czar's time
-    // when only a couple of threads can effective write to a single table.
-    util::ThreadPool::Ptr _largeResultPool = util::ThreadPool::newThreadPool(3, nullptr);
+    // The limited size pool will keep large queries from using up all the czar's time.
+    static util::ThreadPool::Ptr _largeResultPool;
+    static std::mutex largeResultPoolMutex;
```

Comment:
Underscore prefix still needed for this name? I somewhat prefer not to pollute class declaration with static members but hide them in anonymous namespace.